doc/specs/#16599 - Quick Fix.md

```diff
@@ -0,0 +1,43 @@
+---
+author: Carlos Zamora @carlos-zamora
+created on: 2024-04-01
+last updated: 2024-04-02
+issue id: "#16599"
+---
+# Quick Fix
+## Solution Design
+Quick Fix will be a new UI surface that allows the user to interact with the terminal and leverage the context of a specific command being executed. The UI is discussed in the [UI/UX Design section](#ui/ux-design).
+
+The UI itself will live in the TerminalApp project, whereas any logic populating the UI will live in the TerminalControl project. This allows for any consumer of the TerminalControl project to still have access to these features, but have the ability to represent them in whatever UI they please. To accomplish this, the TerminalControl layer must expose a `IVector<String> GetQuickFixContext(UInt32 Row)` which will retrieve all of the quick fix suggestions at the given `Row`. The returned suggestions will be consumed by the TerminalApp layer, where the `IVector<String>` will be converted to "sendInput" actions and displayed in the quick fix UI.
+
+If and when the buffer resizes and reflows, the TerminalApp layer will be notified that the "contexts" are now invalidated. At that point, TerminalApp will call `GetQuickFixContext()` again to retrieve the updated information.
```

Comment:
Admittedly, this is a bit ambiguous and that was kind of intentional 😅. I took a look at how reflow is done for scroll marks and looked through the current resize/reflow code. It looks like the plan there is to retrieve all the scroll marks again, which works great for that scenario!

The problem here is that we actually have to do some work to convert the OSC sequence into a list of WinGet string suggestions. Yes, we'll do that once, but I'd like it if we didn't have to do it again on a resize/reflow. Ideally, we'd have a way of saying "move the quick fix to row X". TerminalApp doesn't have really any knowledge of resize/reflows, it seems, other than updating the pane's size.

So the question for the reviewer: how can we reuse the same `TermControl::QuickFixContext` (or maybe we just reuse something like `TermApp::QuickFixMetadata`?) without having to re-retrieve the WinGet suggestions?

@zadjii-msft Thoughts?

P.S.: Alternatively, the layering abstraction could be wrong here and maybe this is an argument for everything living in TermControl. But that doesn't sound right to me either. 🤔

Reply:
https://github.com/microsoft/terminal/pull/16937 re-writes how marks are done, and how we reflow them. Since we're really just stashing quick fixes on the ROW (we don't care about the x-position within the ROW), then something like what I did with the `ScrollbarData` on ROW is probably all we need

Reply:
but also I agree that it should just live in TermControl so that we don't need to worry about having the control invalidate the app when it scrolls - that way leads madness